packages/api/src/external/commonwell/patient.ts

```diff
@@ -433,51 +433,49 @@ async function validateCWEnabled({
   patient,
   facilityId,
   forceCW,
-  debug,
+  log,
 }: {
   patient: Patient;
   facilityId: string;
   forceCW: boolean;
-  debug: typeof console.log;
+  log: typeof console.log;
 }): Promise<boolean> {
   const { cxId } = patient;
   const isSandbox = Config.isSandbox();
 
   if (forceCW || isSandbox) {
-    debug(`CW forced, proceeding...`);
+    log(`CW forced, proceeding...`);
     return true;
   }
 
   try {
-    const isCwQueryEnabled = await isFacilityEnabledToQueryCW(facilityId, patient);
-    const isCWEnabled = await isCommonwellEnabled();
-    const isEnabledForCx = await isCWEnabledForCx(cxId);
-
-    const cwIsDisabled = !isCWEnabled;
-    const cwIsDisabledForCx = !isEnabledForCx;
-
-    if (cwIsDisabledForCx) {
-      debug(`CW disabled for cx ${cxId}, skipping...`);
+    const [isCwEnabledGlobally, isCwEnabledForCx] = await Promise.all([
+      isCommonwellEnabled(),
+      isCWEnabledForCx(cxId),
+    ]);
+    if (!isCwEnabledGlobally) {
+      log(`CW not enabled, skipping...`);
       return false;
-    } else if (cwIsDisabled) {
-      debug(`CW not enabled, skipping...`);
+    }
+    if (!isCwEnabledForCx) {
+      log(`CW disabled for cx ${cxId}, skipping...`);
       return false;
-    } else if (!isCwQueryEnabled) {
-      debug(`CW not enabled for query, skipping...`);
+    }
+    const isCwQueryEnabled = await isFacilityEnabledToQueryCW(facilityId, patient);
```

Comment:
This is expensive, so we can avoid executing this if CW is not enabled for the cx.